### PR TITLE
E1718 - Review requirements and thresholds

### PR DESCRIPTION
**Pull request by crajase**

### CODE REVIEW

**Comment by Winbobob** [comment]
Hi team,

Please try to fix the issues (except `Cyclomatic complexity`, `Perceived complexity` and `Assignment Branch Condition size`) reported by code climate.

Thanks,
Zhewei

File: app/views/assignments/edit/_review_strategy.html.erb [hunk 1]
```diff
@@ -79,3 +104,35 @@
   </tr>
 
 </table>
+<script>
```
Comment (Winbobob) [inline app/views/assignments/edit/_review_strategy.html.erb]: Insert javascript in the view is not a a good idea.

File: app/models/review_assignment.rb [hunk 2]
```diff
@@ -162,6 +154,14 @@ def reject_by_no_topic_selection_or_no_submission(contributor_set)
     contributor_set
   end
 
+  def filter_least_reviewed(contributor_set)
```
Comment (Winbobob) [inline app/models/review_assignment.rb]: This method `filter_least_reviewed` is exactly the same as `reject_by_least_reviewed`. Why did you rewrite it?

File: app/models/assignment.rb [hunk 3]
```diff
@@ -72,6 +73,15 @@ def has_teams?
     @has_teams ||= !self.teams.empty?
   end
 
+  def valid_num_review
```
Comment (Winbobob) [inline app/models/assignment.rb]: It seems that this method is not been called anywhere.

File: spec/features/review_assignment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
```
Comment (Winbobob) [inline spec/features/review_assignment_spec.rb]: This test suite seems to be duplicated with peer review feature tests.

File: spec/models/assignment_spec.rb [hunk 5]
```diff
@@ -22,6 +22,35 @@
   it "checks whether signed up topic is created or not" do
     expect(create(:topic)).to be_valid
   end
+
+  it "checks whether num_reviews_allowed and num_reviews_required is valid or not" do
```
Comment (Winbobob) [inline spec/models/assignment_spec.rb]: Newly-added tests did not include invalid cases.

File: db/migrate/017_create_assignments.rb [hunk 6]
```diff
@@ -9,8 +9,8 @@ def self.up
     t.column "course_id", :integer, :default => 0, :null => false
     t.column "instructor_id", :integer, :default => 0, :null => false
     t.column "private", :boolean, :default => false, :null => false
-    t.column "num_reviews", :integer, :default => 0, :null => false
-    t.column "num_review_of_reviews", :integer, :default => 0, :null => false
+    t.column "num_reviews", :integer, :default => 3, :null => false
```
Comment (Winbobob) [inline db/migrate/017_create_assignments.rb]: Why did you change this migration file?

File: db/migrate/20170322052959_add_num_reviews_to_assignments.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class AddNumReviewsToAssignments < ActiveRecord::Migration
+  def change
+    change_column :assignments, :num_reviews, :integer, :default => 3
```
Comment (yangsong8) [inline db/migrate/20170322052959_add_num_reviews_to_assignments.rb]: We have this field in assignments table.

Comment (yangsong8) [inline db/migrate/20170322052959_add_num_reviews_to_assignments.rb]: You changed the default value of the field num_reviews, so the class name should not be "Add...."

File: db/migrate/20170322074046_add_num_metareviews_required_to_assignments.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class AddNumMetareviewsRequiredToAssignments < ActiveRecord::Migration
+  def change
+    add_column :assignments, :num_metareviews_required, :integer, :default => 3
```
Comment (yangsong8) [inline db/migrate/20170322074046_add_num_metareviews_required_to_assignments.rb]: Ditto
